### PR TITLE
[DOCS] Migration guide: link to "What's new" page for the same version

### PR DESCRIPTION
**Pull request by abdonpijpelink**
Currently, ES migration guides for older versions link to the "What's new" page of the documentation version, rather than the version you'd like to migrate to. For example, the "Migrating to 8.4" page in the current docs links to the "What's new in 8.6" doc rather than the "What's new in 8.4" doc (see screenshot below). 

This PR should fix that for future versions.

image

### CODE REVIEW

File: build-tools-internal/src/main/resources/templates/breaking-changes.asciidoc [hunk 1]
```diff
@@ -7,7 +7,7 @@
 This section discusses the changes that you need to be aware of when migrating
 your application to {es} ${majorDotMinor}.
 
-See also <<release-highlights>> and <<es-release-notes>>.
+See also {ref-bare}/${majorDotMinor}/release-highlights.html[What's new in ${majorDotMinor}] and <<es-release-notes>>.
```
Comment (lcawl) [inline build-tools-internal/src/main/resources/templates/breaking-changes.asciidoc:10]: > See also {ref-bare}/${majorDotMinor}/release-highlights.html[What's new in ${majorDotMinor}] and <<es-release-notes>>.

Another option to consider might be to add a "next-major" attribute to the docs repo too (e.g. like the prev-major-* attributes we already have in https://github.com/elastic/docs/blob/master/shared/versions/stack/8.6.asciidoc.  That way it'd be controlled uniformly across the library and this URL could just be  "{ref-bare}/{next-major}/..." or something along those lines.

Comment (lcawl) [reply build-tools-internal/src/main/resources/templates/breaking-changes.asciidoc:10]: Ah, nevermind I see that there are multiple "migrating to x.y" in each version, so my suggestion wouldn't solve the problem here.

**Review by lcawl — APPROVED** [review]
Thanks for catching this!

**Comment by abdonpijpelink** [comment]
@elasticmachine update branch

**Pull request merged**